src/Concerns/Arguments/ManagesArguments.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pest\Parallel\Concerns\Arguments;
+
+trait ManagesArguments
+{
+    /**
+     * @var array<int, string>
+     */
+    protected $arguments;
+
+    final public function handle(array $arguments): array
+    {
+        $this->arguments = $arguments;
+        $this->editArguments();
+
+        return $this->arguments;
+    }
+
+    abstract protected function editArguments(): void;
+
+    /**
+     * Add an argument and an optional value to the given arguments.
+     */
+    final private function setArgument(string $key, string $value = ''): self
+    {
+        $argument = $key;
+
+        if ($value !== '') {
+            $argument .= "={$value}";
+        }
+
+        $this->arguments[] = $argument;
+
+        return $this;
+    }
+
+    /**
+     * Remove the given argument from the array if it exists.
+     */
+    final private function unsetArgument(string $argument): self
```

Comment:
Declaring a class method `final private` does not make sense because all `private` class methods are inaccessible by child classes anyway.

Prior to PHP 8.0, it was allowed to declare a class method as `final private`, but it was not allowed to redeclare the method due to `final` clause enforcement.

```php
class Foo {
    final private function testFoo(): void {}
}

class ChildFoo extends Foo{
    private function testFoo(): void {}
}
```

This was not allowed prior to PHP 8.0, and resulted in a fatal error:

`Fatal error: Cannot override final method Foo::testFoo() in ... on line ...`

In PHP 8.0 and later, this restriction is removed, and only a warning is emitted:

`Warning: Private methods cannot be final as they are never overridden by other classes in ... on line ...`

<img width="945" alt="screenshot 170" src="https://user-images.githubusercontent.com/13007665/131880005-bd28ddd0-734f-4275-b776-98b4c3bdc502.png">

[source](https://php.watch/versions/8.0/final-private-function#final-private)

Reply:
Good spot 👌 this is a leftover from refactoring. Will remove when back at my computer